### ISSUE TITLE
Updating select input options results in empty select fields

### ISSUE THREAD

**Issue by mpociot** [livewire/livewire#1337]
### Description

There is a weird behavior when updating select options within a Livewire component and then change the active select option. The select input will be empty in this case.

![livewire](https://user-images.githubusercontent.com/804684/88802773-d5838300-d1ab-11ea-8e28-7e311ae1940e.gif)

### Steps to reproduce

1. Load the Laravel Playground
2. Click "Update Values" - this updates the select input options
3. Change a select option

The select box will be rendered empty, even though the model has the correct model value associated. It works correct for all model changes that come after this.

https://laravelplayground.com/#/snippets/76f36148-f05f-41dd-a6aa-eb1dfa585110

**Context**
- Livewire version: 1.3.2
- Laravel version: 7.22.4
- Browser: Chrome

**Comment by alexandermariduena** [comment]
Hi @mpociot i was able to fix the issue by adding `wire:key`:
```
<select wire:model="selectedValue">
    @foreach($options as $key => $value)
      <option value="{{ $key }}" wire:key="{{ $key }}">{{ $value }}</option>
    @endforeach
</select>
```

Check the following playground:
https://laravelplayground.com/#/snippets/88b991a2-2ab1-4b02-82ac-636074508055

Hope it helps.

**Comment by alexpli** [comment]
I experience the same weird behaviour as @mpociot. Adding `wire:key` didn't solve that for me.

**Comment by ajmariduena** [comment]
Can you reproduce the issue on a playground?.

**Comment by alexpli** [comment]
I found my error. I tried to select the right option in the Livewire `mount` function AND with `<option value="{{ $optionKey }}" @if (old('attribute') === $optionKey) selected @endif>{{ $optionName }}</option>`.

I removed the `<option value="{{ $optionKey }}" @if (old('attribute') === $optionKey) selected @endif>{{ $optionName }}</option>` in the Livewire blade and it works.

**Comment by RoccoHoward** [comment]
Thanks @alexpli - this resolves it for me to. Livewire must be doing the smarts and we're screwing it up.

TLDR; You do not need to put in any if-statement to mark the option selected - Livewire will handle it for you.

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)
